src/builtins.c

```diff
@@ -1643,6 +1643,14 @@ JL_CALLABLE(jl_f__typevar)
 }
 
 // genericmemory ---------------------------------------------------------------------
+JL_CALLABLE(jl_f_memorynew)
+{
+    JL_NARGS(memorynew, 2, 2);
+    JL_TYPECHK(memorynew, datatype, args[0]);
```

Comment:
we should probably also check that it is a GenericMemory subtype
```suggestion
    jl_datatype_t *jl_genericmemory_type_type = jl_datatype_type;
    JL_TYPECHK(memorynew, genericmemory_type, args[0]);
```